localstack/services/sns/sns_listener.py

```diff
@@ -249,28 +243,46 @@ def return_response(self, method, path, data, headers, response):
     def should_persist(self, method, path, data, headers, response):
         req_params = parse_request_data(method, path, data)
         action = req_params.get('Action', '')
-        if action in SKIP_PERSISTENCE_ACTIONS:
+        if action in sns_backend.skip_persistence_actions:
             return False
         return super(ProxyListenerSNS, self).should_persist(method, path, data, headers, response)
 
 
+def patch_moto():
+    def patch_create_platform_endpoint(self, *args):
+        try:
+            return create_platform_endpoint_orig(self, *args)
+        except DuplicateSnsEndpointError:
+            custom_user_data, token = args[2], args[3]
+            for endpoint in self.platform_endpoints.values():
+                if endpoint.token == token:
+                    if custom_user_data and custom_user_data != endpoint.custom_user_data:
+                        raise DuplicateSnsEndpointError('Endpoint already exist for token: %s with different attributes'
+                             % token)
+                    else:
+                        return endpoint
+    create_platform_endpoint_orig = MotoSNSBackend.create_platform_endpoint
+    MotoSNSBackend.create_platform_endpoint = patch_create_platform_endpoint
+
+
+patch_moto()
 # instantiate listener
 UPDATE_SNS = ProxyListenerSNS()
+sns_backend = SNSBackend.get()
```

Comment:
`sns_backend` should not be defined at the global scope. We should call `SNSBackend.get()` inside the request handling methods, as it may return a different instance depending on the request made by the user (i.e., depending on the region defined).